### PR TITLE
feat: edit file writing process

### PR DESCRIPTION
**Pull request by hagoptaminian**
instead of converting retrieved actions to an array via mongodb method, iterate over cursor and write each document to file

### CODE REVIEW

File: controllers/tasks.js [hunk 1]
```diff
@@ -110,47 +110,68 @@ const createTask = [
...
-        await tasksCollection.deleteOne({ _id: task._id });
       }
     });
+
+    // trigger upload/delete/hide/task complete operations
+    try {
+      logger.debug('file write complete; uploading file');
+      const fileId = await uploadFile(
+        // for testing, use spaceId 5ed5f92233c8a33f3d5d87a5
+        `https://graasp.eu/spaces/5ed5f92233c8a33f3d5d87a5/file-upload`,
```
Comment (juancarlosfarah) [inline controllers/tasks.js:163]: This shouldn't be hardcoded, right?

File: services/analytics.js [hunk 2]
```diff
@@ -15,9 +15,10 @@ const fetchActions = async (collection, spaceIds, { sampleSize } = {}) => {
 
   if (sampleSize) {
     aggregateQuery.push({ $sample: { size: sampleSize } });
+    return await collection.aggregate(aggregateQuery).toArray();
```
Comment (juancarlosfarah) [inline services/analytics.js:18]: Can you do the `toArray()` in the function that calls this function? That way this function always returns a cursor and you don't have to worry about inconsistencies.

File: controllers/tasks.js (hunk 1 above)
Comment (hagoptaminian) [reply controllers/tasks.js:163]: 100%. Good catch.

File: controllers/analytics.js [hunk 3]
```diff
@@ -35,9 +35,11 @@ const getAnalytics = async (req, res, next) => {
 
     // fetch (sample of) actions of retrieved space ids
     const spaceIds = spaceTree.map((space) => space.id);
-    const actions = await fetchActions(actionsCollection, spaceIds, {
-      sampleSize: requestedSampleSize,
-    });
+    const actions = await (
```
Comment (juancarlosfarah) [inline controllers/analytics.js:38]: I think you can just `await` on the outer function.

Comment (hagoptaminian) [reply controllers/analytics.js:38]: I'm assuming you mean:

`const actions = await fetchActions(actionsCollection, spaceIds, {sampleSize: requestedSampleSize}).toArray();`

That doesn't work (I tested it). I think: `fetchActions` is asynchronous, so you have to `await` it; it returns the resolved value of `await collection.aggregate(...)` (a cursor), and the `.toArray()` call on that cursor is also asynchronous, so you have to `await` that. Hope I haven't missed anything.

Comment (juancarlosfarah) [reply controllers/analytics.js:38]: If you don't `return await` in line 20 of `/service/analytics` I think it should work. Can you try that?

**Pull request merged**